csharp/src/Google.Protobuf/Reflection/FileDescriptor.cs

```diff
@@ -504,17 +541,16 @@ public override string ToString()
         /// <summary>
         /// The (possibly empty) set of custom options for this file.
         /// </summary>
-        //[Obsolete("CustomOptions are obsolete. Use GetOption")]
+        [Obsolete("CustomOptions are obsolete. Use GetOption")]
```

Comment:
The `[Obsolete]` attribute is something that we should not use. The problem is that for all projects that have "treatWarningsAsErrors" enabled, adding [Obsolete] is basically going to break the build (=breaking change).
Therfore, use of [Obsolete] is discouraged.  Can you replace all ocurrence of [Obsolete] with a simple comment?

Reply:
I disagree with this - I think it's fine to use Obsolete here. It's only a breaking change if users have opted in to treating warnings as errors, at which point they're sort of *asking* for this kind of thing. When they get an error due to this, they can easily fix it by either changing to use GetOption or suppressing this one warning.

So while I agree it's kinda-breaking, I think it's breaking in a "we don't need to take a new major version for this sort of breakage" way.

Reply:
Opinions on this topic differ. I don't feel strongly about this so it's fine to leave as is.